### ISSUE TITLE
Guided Backpropagation ("deconvolution")

### ISSUE THREAD

**Issue by sealedtx** [AlexeyAB/darknet#3539]
Hello, Alexey!
Do you have in plans to implement such feature, so it will be possible to visualise what the network has learnt, or which parts of an input image are most characteristic of certain class? It would be great for understanding training results.

Guided Backpropagation consists of performing a backward pass through the network and computing the gradient w.r.t. an input image. In order to visualise which parts of an image are characteristic of an certain class, the activations of the softmax class posterior layer are all set to zero, except the activation for the class of interest, and subsequently the gradient w.r.t. an input image will activate strongest in the areas characteristic of that class.

**Comment by AlexeyAB** [comment]
@sealedtx Hi,

I will think about it.

Can you get URL to paper, source code or detailed explanation?

**Comment by sealedtx** [comment]
@AlexeyAB I haven't found exact paper about Guided Backpropagation(GB), but [this](https://arxiv.org/pdf/1610.02391.pdf.) paper has another approach to visualize CNN compared to GB and lot of interesting references to similar methods 

May be [this](https://github.com/utkuozbulak/pytorch-cnn-visualizations) pytorch implementation could be useful
